### PR TITLE
RDoc-2642 [Python] Client API > Session > Querying > Query overview [Replace C# samples]

### PR DESCRIPTION
**Pull request by reebhub**
Related issue:
[RDoc-2642](https://issues.hibernatingrhinos.com/issue/RDoc-2642/Python-Client-API-Session-Querying-Query-overview-Replace-C-samples)

### CODE REVIEW

File: Documentation/5.4/Raven.Documentation.Pages/client-api/session/querying/how-to-query.python.markdown [hunk 1]
```diff
@@ -0,0 +1,358 @@
+# Query Overview
+
+---
+
+{NOTE: }
+
+* Queries in RavenDB can be written with either of the following:
+    * __LINQ__ - when querying with the session's `query` method.
```
Comment (poissoncorp) [inline Documentation/5.4/Raven.Documentation.Pages/client-api/session/querying/how-to-query.python.markdown:8]: there's no LINQ in Python 
```suggestion
    The session's `query` method - rich API is provided
```

File: Documentation/5.4/Raven.Documentation.Pages/client-api/session/querying/how-to-query.python.markdown [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+* Queries are timed out after a configurable time period.  See [query timeout](../../../server/configuration/database-configuration#databases.querytimeoutinsec).
+
+---
+
+* In this page:
+  * [Queries always provide results using an index](../../../client-api/session/querying/how-to-query#queries-always-provide-results-using-an-index)  
+  * [session.query](../../../client-api/session/querying/how-to-query#session.query)  
+  * [session.advanced.document_query](../../../client-api/session/querying/how-to-query#session.advanced.document_query)  
+  * [session.advanced.raw_query](../../../client-api/session/querying/how-to-query#session.advanced.raw_query)
+  * [Custom methods and extensions for LINQ](../../../client-api/session/querying/how-to-query#custom-methods-and-extensions-for-linq)
```
Comment (poissoncorp) [inline Documentation/5.4/Raven.Documentation.Pages/client-api/session/querying/how-to-query.python.markdown:44]: no linq in python

File: Documentation/5.4/Raven.Documentation.Pages/client-api/session/querying/how-to-query.python.markdown [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+  * Example RQL: &nbsp; `from Employees where id() == "employees/1-A"`  
+    See more examples [below](../../../client-api/session/querying/how-to-query#session.query).
+
+{NOTE/}
+
+{PANEL/}
+
+{PANEL: session.query}
+
+* The simplest way to issue a query is by using the session's `query` method which supports LINQ.
```
Comment (poissoncorp) [inline Documentation/5.4/Raven.Documentation.Pages/client-api/session/querying/how-to-query.python.markdown:166]: no linq here

File: Documentation/5.4/Raven.Documentation.Pages/client-api/session/querying/how-to-query.python.markdown [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+    See more examples [below](../../../client-api/session/querying/how-to-query#session.query).
+
+{NOTE/}
+
+{PANEL/}
+
+{PANEL: session.query}
+
+* The simplest way to issue a query is by using the session's `query` method which supports LINQ.  
+  Both the LINQ method syntax and the LINQ query syntax are supported.
```
Comment (poissoncorp) [inline Documentation/5.4/Raven.Documentation.Pages/client-api/session/querying/how-to-query.python.markdown:167]:
```suggestion
 
```

File: Documentation/5.4/Raven.Documentation.Pages/client-api/session/querying/how-to-query.python.markdown [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+
+* By default, if the page size is not specified, all matching records will be retrieved from the database.
+
+{NOTE/}
+
+{PANEL/}
+
+{PANEL: session.advanced.document_query}
+
+* `document_query` provides a full spectrum of low-level querying capabilities,
```
Comment (poissoncorp) [inline Documentation/5.4/Raven.Documentation.Pages/client-api/session/querying/how-to-query.python.markdown:250]: it's not true - simple session.query gives also full capabilities, we have document_query in python but it ain't something amazing like in C#, it has the same capabilities like the session.query()

File: Documentation/5.4/Raven.Documentation.Pages/client-api/session/querying/how-to-query.python.markdown [hunk 6]
```diff
@@ -0,0 +1,358 @@
...
+
+__Example__:
+
+{CODE-TABS}
+{CODE-TAB:python:RawQuery query_6_1@ClientApi\Session\Querying\HowToQuery.py /}
+{CODE-TABS/}
+
+{PANEL/}
+
+{PANEL: Custom methods and extensions for LINQ}
```
Comment (poissoncorp) [inline Documentation/5.4/Raven.Documentation.Pages/client-api/session/querying/how-to-query.python.markdown:289]: no linq here, but i can recognize these methods. I've implemented most of them

I haven't implemented yet:
- `group_by_array_content`
- `group_by_array_values`

File: Documentation/5.4/Raven.Documentation.Pages/client-api/session/querying/how-to-query.python.markdown [hunk 7]
```diff
@@ -0,0 +1,358 @@
...
+- [suggest_using](../../../client-api/session/querying/how-to-work-with-suggestions)
+
+{PANEL/}
+
+{PANEL: Syntax}
+
+{CODE:python syntax@ClientApi\Session\Querying\HowToQuery.py /}
+
+
+| Parameter | Type | Description |
```
Comment (poissoncorp) [inline Documentation/5.4/Raven.Documentation.Pages/client-api/session/querying/how-to-query.python.markdown:323]: this description don't match methods.
```py
 def query(
                self, source: Optional[Query] = None, object_type: Optional[Type[_T]] = None
            ) -> DocumentQuery[_T]:
                ...

            def query_collection(
                self, collection_name: str, object_type: Optional[Type[_T]] = None
            ) -> DocumentQuery[_T]:
                ...

            def query_index(self, index_name: str, object_type: Optional[Type[_T]] = None) -> DocumentQuery[_T]:
                ...

            def document_query(
                self,
                index_name: str = None,
                collection_name: str = None,
                object_type: Type[_T] = None,
                is_map_reduce: bool = False,
            ) -> DocumentQuery[_T]:
                ...

            # Overloads for querying an index:
            # ================================
            def query_index_type(
                self, index_type: Type[_TIndex], object_type: Optional[Type[_T]] = None
            ) -> DocumentQuery[_T]:
                ...

            def document_query_from_index_type(
                self, index_type: Type[_TIndex], object_type: Type[_T]
            ) -> DocumentQuery[_T]:
                ...

            # RawQuery
            # ================================
            def raw_query(self, query: str, object_type: Optional[Type[_T]] = None) -> RawDocumentQuery[_T]:
                ...
```

File: Documentation/5.4/Raven.Documentation.Pages/client-api/session/querying/how-to-query.python.markdown (hunk 1 above)
Comment (reebhub) [reply Documentation/5.4/Raven.Documentation.Pages/client-api/session/querying/how-to-query.python.markdown:8]: done

File: Documentation/5.4/Raven.Documentation.Pages/client-api/session/querying/how-to-query.python.markdown (hunk 3 above)
Comment (reebhub) [reply Documentation/5.4/Raven.Documentation.Pages/client-api/session/querying/how-to-query.python.markdown:166]: removed

File: Documentation/5.4/Raven.Documentation.Pages/client-api/session/querying/how-to-query.python.markdown (hunk 4 above)
Comment (reebhub) [reply Documentation/5.4/Raven.Documentation.Pages/client-api/session/querying/how-to-query.python.markdown:167]: removed

File: Documentation/5.4/Raven.Documentation.Pages/client-api/session/querying/how-to-query.python.markdown (hunk 5 above)
Comment (reebhub) [reply Documentation/5.4/Raven.Documentation.Pages/client-api/session/querying/how-to-query.python.markdown:250]: done

File: Documentation/5.4/Raven.Documentation.Pages/client-api/session/querying/how-to-query.python.markdown (hunk 6 above)
Comment (reebhub) [reply Documentation/5.4/Raven.Documentation.Pages/client-api/session/querying/how-to-query.python.markdown:289]: - changed the title and removed the two unimplemented methods  
- please remember to use the "python documentation required" label when you implement them so their description can be readded here

File: Documentation/5.4/Raven.Documentation.Pages/client-api/session/querying/how-to-query.python.markdown (hunk 7 above)
Comment (poissoncorp) [reply Documentation/5.4/Raven.Documentation.Pages/client-api/session/querying/how-to-query.python.markdown:317]: i thought we're going to describe each method, and as python lacks overrides we'd need to describe each `method-with-different-name(override)`

**Review by poissoncorp — APPROVED** [review]

**Pull request merged**